src/sparseml/pytorch/torchvision/train.py

```diff
@@ -323,9 +324,20 @@ def collate_fn(batch):
         steps_per_eval = min(steps_per_eval, args.max_eval_steps)
 
     print("Creating model")
-    model = torchvision.models.get_model(
-        args.model, weights=args.weights, num_classes=num_classes
-    )
+    if args.arch_key in ModelRegistry.available_keys():
+        with torch_distributed_zero_first(args.rank if args.distributed else None):
```

Comment:
Let's explain why: the first process is downloading the model from sparsezoo (other processes need to wait until that's done)